edb/pgsql/metaschema.py

```diff
@@ -236,6 +258,122 @@ def __init__(self) -> None:
         )
 
 
+class StrToConfigMemoryFunction(dbops.Function):
+    """An implementation of std::str to cfg::memory cast."""
+    text = r'''
+        SELECT
+            CASE
+                WHEN m.v[1] IS NOT NULL AND m.v[2] IS NOT NULL
+                THEN (
+                    CASE
+                        WHEN m.v[2] = 'B'
+                        THEN m.v[1]::int8
+
+                        WHEN m.v[2] = 'KiB'
+                        THEN m.v[1]::int8 * 1024
+
+                        WHEN m.v[2] = 'MiB'
+                        THEN m.v[1]::int8 * 1024 * 1024
+
+                        WHEN m.v[2] = 'GiB'
+                        THEN m.v[1]::int8 * 1024 * 1024 * 1024
+
+                        WHEN m.v[2] = 'TiB'
+                        THEN m.v[1]::int8 * 1024 * 1024 * 1024 * 1024
+
+                        WHEN m.v[2] = 'PiB'
+                        THEN m.v[1]::int8 * 1024 * 1024 * 1024 * 1024 * 1024
+
+                        ELSE
+                            -- Won't happen but we still have a guard for
+                            -- completeness.
+                            edgedb.raise(
+                                NULL::edgedb.memory_t,
+                                msg => (
+                                    'unsupported memory size unit' ||
+                                    m.v[2] || '"'
+                                )
+                            )
+                    END
+                )
+                ELSE
+                    edgedb.raise(
+                        NULL::edgedb.memory_t,
+                        msg => (
+                            'unable to parse memory size "' || "val" || '"'
+                        )
+                    )
+            END
+        FROM LATERAL (
+            SELECT regexp_match(
```

Comment:
IMO, the error in the `CASE` block is clearer, so I'd change the unit parsing here to `\w+`

Reply:
Fixed in https://github.com/edgedb/edgedb/pull/3103/commits/97fc60a15629030bda11822898e962b41afaec99